lib/engine/game/g_1849/step/buy_token.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1849
+      module Step
+        class BuyToken < Engine::Step::Base
+          ACTIONS = %w[buy_token pass].freeze
+          MIN_PRICE = 1
+
+          def actions(entity)
+            return [] unless entity == current_entity
+            return [] unless can_buy_token?(entity)
```

Comment:
Can be one line `return [] if entity != current_entity || !can_buy_token?(entity)`

Reply:
And actually, this doesn't make sense in conjunction with the auto action.

Reply:
steps that utilize auto actions don't have an early return in `actions`. They always make the possible actions available and the auto action logic chooses pass for you if you can't act.